src/pixel/mod.rs

```diff
@@ -10,6 +10,15 @@ pub use self::gamma_rgb::GammaRgb;
 mod srgb;
 mod gamma_rgb;
 
+///A transfer function to and from linear space.
+pub trait TransferFn {
```

Comment:
It's a bit hard to generalize that in the `TransferFn` trait, since there are other parameters involved. There's currently the `Srgb::from_linear(color)` method, which should contain all the necessary information to convert anything to `Srgb`.

`let rgb = linear_rgb.from_linear()`, which I guess is the same as `let rgb = linear_rgb.into_nonlinear()`, will always need extra information about the transfer function. That can't be changed by changing the `TransferFn` trait.